stock_picking_customer_ref/models/stock_picking.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2015 AvanzOSC - Alfredo de la Fuente
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    client_order_ref = fields.Char(
+        string="Customer Reference",
+        related="sale_id.client_order_ref",
+        store=True)
```

Comment:
@bizzappdev I think this should be readonly to avoid security and writes problems

Reply:
@RobinetDenisAcsone  Related fields are readonly by default. (https://github.com/odoo/odoo/pull/27236)

Reply:
@bizzappdev Yes, forgot we were in v12 :smile: